packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+  /// provided to a [InlineSpan] object. It defines a `BuzzingText` widget which
+  /// uses the [HapticFeedback] class to vibrate the device when the user
+  /// long-presses the "find the" span, which is underlined in wavy green. The
+  /// hit-testing is handled by the [RichText] widget.
+  ///
+  /// ```dart
+  /// class BuzzingText extends StatefulWidget {
+  ///   @override
+  ///   _BuzzingTextState createState() => _BuzzingTextState();
+  /// }
+  ///
+  /// class _BuzzingTextState extends State<BuzzingText> {
+  ///   LongPressGestureRecognizer _longPressRecognizer;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     _longPressRecognizer = LongPressGestureRecognizer()
+  ///       ..onLongPress = _handlePress;
+  ///   }
+  ///
+  ///   @override
+  ///   void dispose() {
+  ///     _longPressRecognizer.dispose();
+  ///     super.dispose();
+  ///   }
+  ///
+  ///   void _handlePress() {
+  ///     HapticFeedback.vibrate();
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return RichText(
+  ///       text: TextSpan(
+  ///         text: 'Can you ',
+  ///         style: TextStyle(color: Colors.black),
+  ///         children: <InlineSpan>[
+  ///           TextSpan(
+  ///             text: 'find the',
+  ///             style: TextStyle(
+  ///               color: Colors.green,
+  ///               decoration: TextDecoration.underline,
+  ///               decorationStyle: TextDecorationStyle.wavy,
+  ///             ),
+  ///             recognizer: _longPressRecognizer,
+  ///           ),
+  ///           TextSpan(
+  ///             text: ' secret?',
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  /// {@end-tool}
+  final GestureRecognizer recognizer;
+
+  /// An alternative semantics label for this text.
+  ///
+  /// If present, the semantics of this span will contain this value instead
+  /// of the actual text.
+  ///
+  /// This is useful for replacing abbreviations or shorthands with the full
+  /// text value:
+  ///
+  /// ```dart
+  /// TextSpan(text: r'$$', semanticsLabel: 'Double dollars')
+  /// ```
+  final String semanticsLabel;
+
+  /// Utility method to convert InlineSpan into [TextSpan] or [WidgetSpan].
+  ///
+  /// Will return null if span is not of type T or null. Otherwise, return
+  /// the span cast as T. Values of T that are not subclasses of [InlineSpan]
+  /// will result in null.
+  static T asType<T>(InlineSpan span) {
+    if (span is! T)
+      return null;
+    return span as T;
+  }
+
+  /// Apply the properties of this object to the given [ParagraphBuilder], from
+  /// which a [Paragraph] can be obtained. [Paragraph] objects can be drawn on
+  /// [Canvas] objects.
+  void build(ui.ParagraphBuilder builder, { double textScaleFactor = 1.0, List<PlaceholderDimensions> dimensions });
```

Comment:
the `dimensions` argument is dubious... it's passing information specific to one kind of InlineSpan, but it's in the interface for all InlineSpans. It would be cleaner if the code related to dimensions was all in the PlaceholderSpan, but I'm not sure how to do this here...